nsqd/http.go

```diff
@@ -226,6 +231,32 @@ func (s *httpServer) getTopicFromQuery(req *http.Request) (url.Values, *Topic, e
 	return reqParams, s.ctx.nsqd.GetTopic(topicName), nil
 }
 
+func (s *httpServer) doLookup(req *http.Request) (interface{}, error) {
```

Comment:
We should be able to share this code (and the implementation for `/topics`, `/channels`, and `/nodes`) across `nsqd` and `nsqlookupd` if we write this as a function that takes in the registration db:

```go
func (s *httpServer) doLookup(req *http.Request, rdb *registrationdb.RegistrationDB) (interface{}, error) {
...
}
```

Adding the other two makes this close to sufficient to back nsqadmin (though we should probably namespace these endpoints under `/lookup/...`).